apps/passport-client/src/validateState.ts

```diff
@@ -175,14 +184,29 @@ export function getRunningAppStateValidationErrors(
     errors.push("missing 'identity'");
   }
 
+  if (!identityV4) {
+    errors.push("missing 'identityV4'");
+  }
+
   const identityFromPCDCollection = identityPCDFromCollection?.claim?.identity;
   const commitmentOfIdentityPCDInCollection =
     identityFromPCDCollection?.commitment?.toString();
   const commitmentFromSelfField = self?.commitment;
   const commitmentFromIdentityField = identity?.commitment?.toString();
 
+  const commitmentOfIdentityV4PCDInCollection =
+    identityPCDFromCollection?.claim.identityV4?.commitment?.toString();
+  const commitmentV4FromSelfField = self?.semaphore_v4_commitment;
+  const commitmentV4FromIdentityField = identityV4?.commitment?.toString();
+  const publicKeyV4FromSelfField = self?.semaphore_v4_pubkey;
+  const publicKeyV4FromIdentityField = identityV4 && v4PublicKey(identityV4);
+
   if (commitmentFromSelfField === undefined) {
-    errors.push(`'self' missing a commitment`);
+    errors.push(`'self' missing a v3 commitment`);
+  }
+
+  if (commitmentV4FromSelfField === undefined) {
```

Comment:
This is a valid case when starting up offline, if the upgrade + update self isn't able to complete.

Reply:
Good point. Fixed.